packages/frontend-core/src/api/index.ts

```diff
@@ -174,29 +151,31 @@ export const createAPIClient = config => {
       })
     } catch (error) {
       delete cache[url]
-      throw makeError("Failed to send request", { url, method })
+      throw makeError("Failed to send request", url, method)
     }
 
     // Handle response
     if (response.status >= 200 && response.status < 400) {
       handleMigrations(response)
       try {
-        if (parseResponse) {
+        if (response.status === 204) {
+          return undefined as ResponseT
+        } else if (parseResponse) {
           return await parseResponse(response)
         } else {
-          return await response.json()
+          return (await response.json()) as ResponseT
         }
       } catch (error) {
         delete cache[url]
-        return null
+        throw `Failed to parse response: ${error}`
```

Comment:
Do you know the reason why we were not throwing, before?

Reply:
I think it was probably just a mistake from before, but I don't honestly know... We need to throw now with TS, since we can't return the proper type when we hit this - but yea I think we probably should have been always throwing, for correctness.